### ISSUE TITLE
Multiple Clusters in same project Garbage collect each others Elastic IPs

### ISSUE THREAD

**Issue by c0dyhi11** [equinix/cloud-provider-equinix-metal#154]
I have two Clusters in the same EQM project. One in Dallas and one in Sunnyvale. 

If I create a service in the project in Dallas, the IP is received and BGP starts... 
And within a minute or so the IP is deleted by the cluster in Sunnyvale. 

I am using an older version of the CCM, so if this has been addressed in a newer version please tell me.

***Dallas Cluster Logs (Create)***
```
2021/02/26 19:20:07 [DEBUG] GET https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips?
2021/02/26 19:20:08 [DEBUG] POST https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips
I0226 19:20:19.754757       1 event.go:281] Event(v1.ObjectReference{Kind:"Service", Namespace:"default", Name:"testvm-ssh", UID:"7fb57fc6-13e2-43d2-93e9-4600c5f5156d", APIVersion:"v1", ResourceVersion:"5705385", FieldPath:""}): type: 'Normal' reason: 'LoadbalancerIP'  -> 139.178.86.116
E0226 19:20:19.755379       1 cloud.go:247] failed to update and sync service for add default/testvm-ssh: elastic ip tag is empty. Nothing to do
E0226 19:20:19.756449       1 event.go:263] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"testvm-ssh.1667626ede823bfd", GenerateName:"", Namespace:"default", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Service", Namespace:"default", Name:"testvm-ssh", UID:"7fb57fc6-13e2-43d2-93e9-4600c5f5156d", APIVersion:"v1", ResourceVersion:"5705385", FieldPath:""}, Reason:"LoadbalancerIP", Message:" -> 139.178.86.116", Source:v1.EventSource{Component:"service-controller", Host:""}, FirstTimestamp:v1.Time{Time:time.Time{wall:0xc006705cecfa7dfd, ext:9782569688704, loc:(*time.Location)(0x284fbe0)}}, LastTimestamp:v1.Time{Time:time.Time{wall:0xc006705cecfa7dfd, ext:9782569688704, loc:(*time.Location)(0x284fbe0)}}, Count:1, Type:"Normal", EventTime:v1.MicroTime{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, Series:(*v1.EventSeries)(nil), Action:"", Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events is forbidden: User "system:serviceaccount:kube-system:cloud-controller-manager" cannot create resource "events" in API group "" in the namespace "default"' (will not retry!)
2021/02/26 19:20:23 [DEBUG] GET https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips?
2021/02/26 19:20:24 [DEBUG] GET https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips?
```

***Sunnyvale Cluster Logs (Delete)***
```
2021/02/26 19:20:43 [DEBUG] GET https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips?
2021/02/26 19:20:44 [DEBUG] GET https://api.packet.net/projects/0e1abd9d-2c7e-4f81-a6d2-7ecac42c9379/ips?
2021/02/26 19:20:45 [DEBUG] DELETE https://api.packet.net/ips/13976351-2713-43e6-ba01-90c8234a9acb
```

***Kubernetes Version:*** v1.18.6-gke.6600
***CCM Version:*** v2.0.0
***Kube-Vip Version:*** v0.2.3

**Comment by displague** [comment]
The IP addresses may need to include additional tags (or use IP customdata) to include some kind of cluster identifier.  The CCM can then ignore any IPs that are not tagged for that cluster.

We would need to be careful about existing IP ranges that are not currently cluster tagged, fixing these on the next reconcile if the cluster tag is missing but other well-known tags are present.

Additionally, and perhaps as a quick fix, we could have the CCM check the facility for each IP range and ignore ranges that belong to a different facility.  If two clusters were in the same project and the same facility we might want to use the cluster tagging approach to prevent clusters from fighting over ranges.  

It may be beneficial to share available ranges between multiple clusters in the same project + facility.

**Comment by deitch** [comment]
I missed this issue @c0dyhi11 .

Yes, this is fixed as of v3.0.0. See #123 

Please test and we can close.

**Comment by c0dyhi11** [comment]
Sorry for not testing this yet. Anthos doesn't yet allow the proper `provider-id` fields that the CCM requires. So I can't test this as is. I'm still using a modified version of v2.0.0
